metricflow/model/validations/reserved_keywords.py

```diff
@@ -0,0 +1,162 @@
+                        ),
+                        message=f"'{dimension.name}' is an SQL reserved keyword, and thus cannot be used as a dimension 'name'.",
+                    )
+                )
+
+        for identifier in data_source.identifiers:
+            if identifier.is_composite:
+                msg = "'{name}' is an SQL reserved keyword, and thus cannot be used as a sub-identifier 'name'"
+                names = [
+                    sub_ident.name for sub_ident in identifier.identifiers if sub_ident.expr is None and sub_ident.name
+                ]
+            else:
+                msg = "'{name}' is an SQL reserved keyword, and thus cannot be used as an identifier 'name'"
+                names = [identifier.name]
+
+            for name in names:
+                if name.upper() in RESERVED_KEYWORDS:
+                    issues.append(
+                        ValidationError(
+                            context=DataSourceElementContext(
+                                file_context=FileContext.from_metadata(data_source.metadata),
+                                data_source_element=DataSourceElementReference(
+                                    data_source_name=data_source.name, element_name=identifier.name
+                                ),
+                                element_type=DataSourceElementType.IDENTIFIER,
+                            ),
+                            message=msg.format(name=name),
+                        )
+                    )
+
+        for measure in data_source.measures:
+            if measure.name.upper() in RESERVED_KEYWORDS:
+                issues.append(
+                    ValidationError(
+                        context=DataSourceElementContext(
+                            file_context=FileContext.from_metadata(data_source.metadata),
+                            data_source_element=DataSourceElementReference(
+                                data_source_name=data_source.name, element_name=measure.name
+                            ),
+                            element_type=DataSourceElementType.MEASURE,
+                        ),
+                        message=f"'{measure.name}' is an SQL reserved keyword, and thus cannot be used as an measure 'name'.",
+                    )
+                )
+
+        return issues
+
+    @classmethod
+    @validate_safely(whats_being_done="checking that data_source sql_tables are not sql reserved keywords")
+    def _validate_data_sources(cls, model: UserConfiguredModel) -> List[ValidationIssueType]:
+        """Checks names of objects that are not nested."""
+        issues: List[ValidationIssueType] = []
+        set_keywords = set(RESERVED_KEYWORDS)
+
+        for data_source in model.data_sources:
+            if data_source.sql_table is not None:
+                set_sql_table_path_parts = set(
+                    [part.upper() for part in SqlTable.from_string(data_source.sql_table).parts_tuple]
+                )
```

Comment:
Seems like you should be able to just do this:

```suggestion
                set_sql_table_path_parts = set(
                    SqlTable.from_string(data_source.sql_table).parts_tuple
                )
```

Reply:
So `.parts_tuple` doesn't guarantee that the parts are all in upper case. And we're storing the reserved keywords all in upper case. A quick check
```Python
>>> parts = ['select', 'where', 'from']
>>> reserved_keywords = set(['SELECT', 'WHERE', 'FROM'])
>>> lower = set(parts)
>>> upper = set([part.upper() for part in parts])
>>> reserved_keywords.intersection(lower)
set()
>>> reserved_keywords.intersection(upper)
{'SELECT', 'WHERE', 'FROM'}
```
Thus, ensuring things are in upper case is necessary